content/components/nrf52.md

```diff
@@ -71,6 +71,34 @@ There are two ways to reference GPIO pins:
 1. By pin name, e.g., `P0.15` or `P1.11`.
 1. By pin number, e.g., `15` or `43`.
 
+## DFU Component
+
+The ``dfu`` component enables automatic entry into **DFU (Device Firmware Update)** mode by monitoring
+the USB CDC serial connection. When a host opens the port at **1200 baud**, the component triggers
+a reset via a GPIO pin to put the device into DFU mode.
+
+ESPHome uses this component internally when uploading firmware via:
+
+```bash
+esphome upload d.yaml
+```
+
+### Example Configuration
+
+```yaml
+nrf52:
+  dfu:
+    reset_pin:
+      number: 14
+      inverted: true
+      mode:
+        output: true
```

Comment:
output mode is set automatically by the schema used - `pins.gpio_output_pin_schema`
```suggestion
```